src/Features/VisualBasic/Portable/InitializeParameter/VisualBasicInitializeMemberFromParameterCodeRefactoringProvider.vb

```diff
@@ -40,5 +41,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.InitializeParameter
         Protected Overrides Sub InsertStatement(editor As SyntaxEditor, functionDeclaration As SyntaxNode, method As IMethodSymbol, statementToAddAfterOpt As SyntaxNode, statement As StatementSyntax)
             InitializeParameterHelpers.InsertStatement(editor, functionDeclaration, statementToAddAfterOpt, statement)
         End Sub
+
+        Protected Overrides Function CreateFieldHelper(document As Document, functionDeclaration As SyntaxNode, model As SemanticModel, cancellationToken As CancellationToken) As Accessibility
+            Dim accessibilityLevel = Accessibility.[Private]
+            Dim service = document.GetLanguageService(Of ISyntaxFactsService)()
+            Dim containingTypeNode = service.GetContainingTypeDeclaration(functionDeclaration, functionDeclaration.SpanStart)
+
+            If containingTypeNode IsNot Nothing Then
+                Dim containingTypeSymbol = CType(model.GetDeclaredSymbol(containingTypeNode, cancellationToken), INamedTypeSymbol)
+
+                ' We always generate private fields. Fields are public by default, except in the case of classes and modules. In those two cases, we can safely remove the accessibility modifier.
+                Select Case containingTypeSymbol.TypeKind
+                    Case TypeKind.[Class], TypeKind.[Module]
+                        accessibilityLevel = Accessibility.NotApplicable
+                End Select
+            End If
+            Return accessibilityLevel
+        End Function
```

Comment:
I think a better way to do things would be to ahve a helper that answers: "What would the accessibility of this member be if it did *not* have an accessibility modifier".  You could then take that and use that alone (along with teh value of the user option) to determine if ou need to supply the accessibility modifier, or if you can just pass NotApplicable.